src/main/java/org/triplea/client/ui/javafx/util/LoadingTask.java

```diff
@@ -1,9 +1,11 @@
-package org.triplea.client.ui.javafx;
+package org.triplea.client.ui.javafx.util;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
 import java.util.function.Function;
 
+import org.triplea.client.ui.javafx.TripleA;
```

Comment:
Probably just because I don't have a deep enough understanding of your vision for the JFX client, but when I see a "utility" class with a direct dependency on the entry point for the application, I get a faint whiff of a smell.  One of the big issues we have with `GameRunner` is that it became kind of a _God Class_, and it would be nice to avoid that same thing in the JFX client.

We may want to consider breaking the dependency by having `TripleA` pass some kind of callback interface to `LoadingTask`.

Reply:
Agreed.
The dependency is related due to a planned "loading animation" that needs to be invoked on the root UI container.